src/PropertyHandler/UnixTimeExporter.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Crell\Serde\PropertyHandler;
+
+use Crell\Serde\Attributes\Field;
+use Crell\Serde\Attributes\UnixTimeField;
+use Crell\Serde\DeformatterResult;
+use Crell\Serde\Deserializer;
+use Crell\Serde\Serializer;
+
+class UnixTimeExporter implements Importer, Exporter {
+
+    /**
+     * @param Serializer $serializer
+     * @param Field $field
+     * @param \DateTimeInterface $value
+     * @param mixed $runningValue
+     * @return mixed
+     */
+    public function exportValue(Serializer $serializer, Field $field, mixed $value, mixed $runningValue): mixed
+    {
+        /** @var UnixTimeField|null $typeField */
+        $typeField = $field->typeField;
+
+        $multiplier = $typeField?->resolution->value;
+        $timestamp = (float) $value->format('U.u');
+
+        return $serializer->formatter->serializeInt($runningValue, $field, (int) ($timestamp * $multiplier));
```

Comment:
I don't quite follow this.  Wouldn't this mean turning "45.67" into "4567", which is... substantially more seconds than we want?

Or is the intent here to record "the number of milliseconds" rather than "time accurate to milliseconds"?  To me the latter seems like the more useful case, but I don't often do fancy date/time work these days so I'm not certain.

Reply:
The developer specifies the resolution the (de)serialization should respect. So, if they specify milliseconds, that is 1000 milliseconds per second. So, we convert the current timestamp to seconds (float) and then multiply by the current resolution (* 1000 in the case of milliseconds). Then, we truncate it to an integer to ensure that only that many milliseconds as passed. This way, if you are (de)serializing an object that represents unix time in a sepcific resolution, you can represent and parse that value. If you want to serialize and deserialize with different resolutions, than similar to other attributes, you can use `scopes` to change the behavior in various contexts.

> Or is the intent here to record "the number of milliseconds" rather than "time accurate to milliseconds"? To me the latter seems like the more useful case, but I don't often do fancy date/time work these days so I'm not certain.

In the case of unix time, these are effectively the same thing. Per definition, unix time is only measured in whole seconds. In practice, various protocols use different resolutions and are integers. For example, NATs uses nanosecond resolution (represented as whole nanoseconds since the unix epoch), browsers use milliseconds (represented as whole milliseconds since the epoch), unix/linux uses seconds (represented as whole seconds since the epoch), and I'm sure something out there probably uses microsecond resolution, since it was explicetly brought up.

Reply:
Just an update: I've reached out to Derick Rethans to get his take on the best way to handle sub-second resolution.  I figure that's best expert we'll find. 😄   I'll let you know what if anything he recommends.

Reply:
There is a problem with exporting it as `time` * `resolution` here: You can now no longer handle the full original 64-bit second range, but rather, only a 1000th of with *milliseconds*. It's a factor of a thousand worse for microseconds.

There is an additional issue here, which is that Javascript (and hence, JSON) do not support integers at all. It only supports floats. This means that even with seconds, you only have an accurate 53-bit range (53-bits make up the storage for exact integers). Which the multiplication towards milliseconds and microseconds reduce by 10 bits and 20 bits respectively. This means that with your range for microseconds is ~33bits, which means the range is ~ 1835-2105. That can be OK, but probably is not.

---

With nanoseconds, this becomes way worse, as the range would become ~24 bits, which provides a range from `Thu 25 Sep 22:49:52 BST 1969` to `Wed  8 Apr 03:10:08 BST 1970`. Definitely not enough.

It is therefore probably better, to not store them in one field.

Reply:
I'm not sure I follow.

The correct numbers appear to be rendered correctly (I'm able to render up to 2262ish with nanosecond resolution), and JSON and JavaScript aren't the only formats and languages interacted with. Serde can also serialize to an array (my primary use-case) and CSV.

Since my initial work on this PR, I've started working with NATs, which use nanosecond timestamps. It'd be nice to use this library with it.

Reply:
Can DateTime even handle nanoseconds?  How would you even represent that in PHP?

Reply:
No, it can only represent up to microseconds (unless you use Carbon). The important part is that we can output/read a format in nanoseconds. A loss of precision is inevitable here -- but there's nothing we can really do there. If someone wants to keep the precision, they probably shouldn't be serializing it as a datetime (and they probably aren't using PHP).

NATs uses nanosecond timestamps, and I know python uses also uses nanosecond timestamps (time_ns, IIRC).

Reply:
How often will you be sending data TO a NAT from PHP?  If you're reading from a NAT, the incoming data could be in nanoseconds-since-epoc, but that would still get folded up to ms by DateTime.  And... I cannot figure out when it would even make sense to send data *to* a NAT with that resolution.

Reply:
I'm totally all for not supporting nanosecond resolution (is IS non-sensical in PHP, after all), though it will only be a matter of time (weeks/years/decades) before someone comes along and needs to consume a message in nanosecond resolution.

That being said, there are only a couple of practical resolutions IMO: milliseconds and seconds. Seconds for regular timestamps and milliseconds for interactions with the browser. Anything other than that is an edge case or potentially "non-standard."